zilliqa/src/consensus.rs

```diff
@@ -1335,7 +1372,17 @@ impl Consensus {
 
     /// Check the validity of a block
     fn check_block(&mut self, block: &Block) -> Result<()> {
-        block.verify_hash()?;
+        let Some(parent) = self.get_block(&block.parent_hash())? else {
```

Comment:
What was the reason for moving the parent check above the `block.view == 0` check?
If you think it's useful, then I think a comment should be added to the function that it cannot be used for genesis, and the `if block.view() == 0` block below can be deleted outright.

But I think having that short-circuit in case it's ever called for genesis is a nice thing to have. If were to want to prevent `check_block` from ever being applied to genesis, then an explicit check returning an dedicated error (rather than `MissingBlockError`) would be better.